packages/wrangler/src/publish.ts

```diff
@@ -78,10 +79,12 @@ export default async function publish(props: Props): Promise<void> {
 
   let file: string;
   if (props.script) {
-    file = props.script;
+    // If the script name comes from the command line it is relative to the current working directory.
+    file = path.resolve(props.script);
   } else {
+    // If the script name comes from the config, then it is relative to the wrangler.toml file.
```

Comment:
good catch!

Reply:
It took me about two hours to get my head around all these different paths. See also the bit below where we have to resolve the entryPoint paths in the `metafile.outputs` relative to `destination.path`!

Reply:
madness!